x-pack/plugins/security_solution/server/lib/telemetry/endpoint_task.ts

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import { Logger } from 'src/core/server';
+import {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '../../../../task_manager/server';
+import { TelemetryEventsSender } from './sender';
+import { FullAgentPolicyInput } from '../../../../fleet/common/types/models/agent_policy';
+import {
+  EndpointMetricsAggregation,
+  EndpointPolicyResponseAggregation,
+  EndpointPolicyResponseDocument,
+  FleetAgentCacheItem,
+} from './types';
+
+export const TelemetryEndpointTaskConstants = {
+  TIMEOUT: '5m',
+  TYPE: 'security:endpoint-meta-telemetry',
+  INTERVAL: '24h',
+  VERSION: '1.0.0',
+};
+
+export class TelemetryEndpointTask {
+  private readonly logger: Logger;
+  private readonly sender: TelemetryEventsSender;
+
+  constructor(
+    logger: Logger,
+    taskManager: TaskManagerSetupContract,
+    sender: TelemetryEventsSender
+  ) {
+    this.logger = logger;
+    this.sender = sender;
+
+    taskManager.registerTaskDefinitions({
+      [TelemetryEndpointTaskConstants.TYPE]: {
+        title: 'Security Solution Telemetry Endpoint Metrics and Info task',
+        timeout: TelemetryEndpointTaskConstants.TIMEOUT,
+        createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+          const { state } = taskInstance;
+
+          return {
+            run: async () => {
+              const lastExecutionTimestamp = moment().utc().toISOString();
+
+              const hits = await this.runTask(taskInstance.id);
+              this.logger.debug(`hits: ${hits}`);
+
+              return {
+                state: {
+                  lastExecutionTimestamp,
+                  runs: (state.runs || 0) + 1,
+                },
+              };
+            },
+            cancel: async () => {},
+          };
+        },
+      },
+    });
+  }
+
+  public start = async (taskManager: TaskManagerStartContract) => {
+    try {
+      await taskManager.ensureScheduled({
+        id: this.getTaskId(),
+        taskType: TelemetryEndpointTaskConstants.TYPE,
+        scope: ['securitySolution'],
+        schedule: {
+          interval: TelemetryEndpointTaskConstants.INTERVAL,
+        },
+        state: { runs: 0 },
+        params: { version: TelemetryEndpointTaskConstants.VERSION },
+      });
+    } catch (e) {
+      this.logger.error(`Error scheduling task, received ${e.message}`);
+    }
+  };
+
+  private getTaskId = (): string => {
+    return `${TelemetryEndpointTaskConstants.TYPE}:${TelemetryEndpointTaskConstants.VERSION}`;
```

Comment:
Did we decide that we need to version this vs. just clobber going forward? I forget what we said.

Reply:
We settled on versioning to keep our options open with user upgrades.